public_api/src/registry_types.ts

```diff
@@ -220,6 +245,12 @@ export interface MachineDefinitionHandlers {
    * or `undefined` to not change anything.
    */
   receive?: MachineCallback<MachineRecieveHandlerArg, number | undefined>;
+
+  /**
+   * Called each time after a Network has finished distributing power
+   * contains information on each category sent in that pass with the starting and remaining budget.
+   */
+  networkStatEvent?: MachineCallback<NetworkStatArg, undefined>;
```

Comment:
- this should be in `MachineDefinitionEvents`, not `MachineDefinitionHandlers`

- use `MachineEventCallback<T>` instead of `MachineCallback<T, undefined>`

- `networkStatEvent` doesn't make sense as an event name. the *event* should be a past tense action. for example, "key pressed" (`onKeyPressed`) or "session created" (`onSessionCreated`). maybe name this event `onNetworkMachineAllocationCompleted` instead